code/mob/living/critter.dm

```diff
@@ -87,6 +87,9 @@ ADMIN_INTERACT_PROCS(/mob/living/critter, proc/modify_health, proc/admincmd_atta
 	var/ghost_spawned = FALSE //Am i inhabited by a ghost player who used the respawn critter option?
 	var/original_name = null
 
+	/// region holder for eat_limb and stomach_retreat
+	var/datum/component/mimic_stomach/stomachHolder = null
```

Comment:
This is not how components are supposed to work. Use signals (or at a stretch, `GetComponent`) to communicate between them, don't hold direct references.